### ISSUE TITLE
Auto-reset permissions

### ISSUE THREAD

**Issue by MuntashirAkon** [MuntashirAkon/AppManager#716]
Android 11 introduced a new feature known as “auto-reset permissions” which keeps track of the unused apps and offer to reset permissions for the selected apps. Recently, Google Play Protect has introduced similar features with backward compatibility from Android 6. Our intention is to take it to the next level (and help the deGoogled users) by introducing similar features in App Manager:

1. Let the user select an interval for outdated apps (default: 3 months but may not work in all platforms) and scan interval (default: no scan)
2. Scan and list outdated apps with dangerous permissions and app ops
3. Provide options to ignore app ops or revoke permissions (or take an user to the settings page if root/ADB is unavailable)

**Comment by Raif1** [comment]
Is there a way to add an option to turn of all the permissions when device is turned off ?

**Comment by MuntashirAkon** [comment]
> Is there a way to add an option to turn of all the permissions when device is turned off ?

No, but you can turn off all permissions after booting is complete. To do this, create a new profile with all the apps whose permissions you need to turn off, set permissions to `*`, and then set up the profile with apps such as Tasker as given in 3ab3c4a9.

**Comment by rodrigoswz** [comment]
@MuntashirAkon I was about to create an issue requesting Tasker integration when I found this mentioned API. In my case, I would like to launch activities from certain apps, something that is possible with [Root Activity Launcher](https://github.com/zacharee/RootActivityLauncher).

Would this be possible with App Manager?

**Comment by MuntashirAkon** [comment]
> @MuntashirAkon I was about to create an issue requesting Tasker integration when I found this mentioned API. In my case, I would like to launch activities from certain apps, something that is possible with [Root Activity Launcher](https://github.com/zacharee/RootActivityLauncher).
> 
> Would this be possible with App Manager?

Via Tasker? Not possible. Via App Manager? Possible. But this is a completely irrelevant to the conversation here and should be asked by creating a separate thread.

**Comment by shuvashish76** [comment]
1, 3. Both sounds like the job of Routine Operations #61 IFTT (via profile?)
2. Job of Permission monitor #1023 and Activity log #143 (i.e. list of apps going to fall under Inactive/Out dated apps based on their AppUsage and last update date etc.)

IMO Auto-reset permissions = one type of event/module for Routine Operations.

**Comment by MuntashirAkon** [comment]
> Auto-reset permissions = one type of event/module for Routine Operations.

It has the label "Routine Ops" for this reason. However, that not all, it is also one of the special filters. That's why it's classified separately.

Ideally, all routine operations should be triggered by a third-party apps (like Automation or Tasker) just as they can be managed from App Manager (but with a potentially good UI).
